Type 'DynamoDBClient' does not satisfy the constraint 'SmithyClient'.

**Describe the bug**
I'm not using the waiters code, but running "tsc" on a project using @aws-sdk/client-dynamodb  throws this error:
```
yarn run v1.22.4
$ tsc
node_modules/@aws-sdk/client-dynamodb/types/waiters/waitForTableExists.d.ts:9:71 - error TS2344: Type 'DynamoDBClient' does not satisfy the constraint 'SmithyClient'.
  The types of 'middlewareStack.concat' are incompatible between these types.
    Type '<InputType extends ServiceInputTypes, OutputType extends ServiceOutputTypes>(from: MiddlewareStack<InputType, OutputType>) => MiddlewareStack<InputType, OutputType>' is not assignable to type '<InputType extends any, OutputType extends any>(from: MiddlewareStack<InputType, OutputType>) => MiddlewareStack<InputType, OutputType>'.
      Types of parameters 'from' and 'from' are incompatible.
        Type 'MiddlewareStack<InputType, OutputType>' is not assignable to type 'MiddlewareStack<InputType, ServiceOutputTypes>'.
          Types of property 'addRelativeTo' are incompatible.
            Type '(middleware: MiddlewareType<InputType, OutputType>, options: RelativeMiddlewareOptions) => void' is not assignable to type '(middleware: MiddlewareType<InputType, ServiceOutputTypes>, options: RelativeMiddlewareOptions) => void'.
              Types of parameters 'middleware' and 'middleware' are incompatible.
                Type 'MiddlewareType<InputType, ServiceOutputTypes>' is not assignable to type 'MiddlewareType<InputType, OutputType>'.
                  Type 'InitializeMiddleware<InputType, ServiceOutputTypes>' is not assignable to type 'MiddlewareType<InputType, OutputType>'.
                    Type 'InitializeMiddleware<InputType, ServiceOutputTypes>' is not assignable to type 'InitializeMiddleware<InputType, OutputType>'.
                      Call signature return types 'InitializeHandler<InputType, ServiceOutputTypes>' and 'InitializeHandler<InputType, OutputType>' are incompatible.
                        Type 'Promise<InitializeHandlerOutput<ServiceOutputTypes>>' is not assignable to type 'Promise<InitializeHandlerOutput<OutputType>>'.
                          Type 'InitializeHandlerOutput<ServiceOutputTypes>' is not assignable to type 'InitializeHandlerOutput<OutputType>'.
                            Types of property 'output' are incompatible.
                              Type 'ServiceOutputTypes' is not assignable to type 'OutputType'.
                                'ServiceOutputTypes' is assignable to the constraint of type 'OutputType', but 'OutputType' could be instantiated with a different subtype of constraint 'any'.
                                  Type 'MetadataBearer' is not assignable to type 'OutputType'.
                                    'MetadataBearer' is assignable to the constraint of type 'OutputType', but 'OutputType' could be instantiated with a different subtype of constraint 'any'.
```

Maybe the SDK doesn't work with older versions of typescript or something.

**SDK version number**

3.0.0

**Is the issue in the browser/Node.js/ReactNative?**

Node.js

**Details of the browser/Node.js/ReactNative version**
Paste output of `npx envinfo --browsers` or `node -v` or `react-native -v`

node is v12.18.2

But maybe more useful is that typescript is 3.8.3


**To Reproduce (observed behavior)**
Given I haven't used the waiters code I wonder if just the include would cause it to fail, but this is approximately my code:

```
import { DynamoDB, GetItemCommand } from '@aws-sdk/client-dynamodb';
import { marshall, unmarshall } from '@aws-sdk/util-dynamodb';

const client = new DynamoDB({ region: 'eu-west-1' });
const command = new GetItemCommand({
    TableName: 'foo',
    Key: marshall({ myKey: "hello" }),
  });
  const { Item } = await client.send(command)
}

```

**Expected behavior**
It compiles.

--- a/packages/util-waiter/package.json
+++ b/packages/util-waiter/package.json
@@ -7,7 +7,6 @@
     "tslib": "^1.8.0"
   },
   "devDependencies": {
-    "@aws-sdk/smithy-client": "3.0.0",
     "@aws-sdk/types": "3.0.0",
     "@types/jest": "^26.0.4",
     "jest": "^26.1.0",

--- a/packages/util-waiter/package.json
+++ b/packages/util-waiter/package.json
@@ -4,10 +4,11 @@
   "description": "Shared utilities for client waiters for the AWS SDK",
   "dependencies": {
     "@aws-sdk/abort-controller": "3.0.0",
-    "@aws-sdk/types": "3.0.0",
     "tslib": "^1.8.0"
   },
   "devDependencies": {
+    "@aws-sdk/smithy-client": "3.0.0",
+    "@aws-sdk/types": "3.0.0",
     "@types/jest": "^26.0.4",
     "jest": "^26.1.0",
     "typescript": "~4.1.2"

--- a/packages/util-waiter/src/createWaiter.ts
+++ b/packages/util-waiter/src/createWaiter.ts
@@ -1,13 +1,8 @@
 import { AbortSignal } from "@aws-sdk/types";
 
 import { runPolling } from "./poller";
-import { sleep, validateWaiterOptions } from "./utils";
-import { SmithyClient, WaiterOptions, WaiterResult, waiterServiceDefaults, WaiterState } from "./waiter";
-
-const waiterTimeout = async (seconds: number): Promise<WaiterResult> => {
-  await sleep(seconds);
-  return { state: WaiterState.TIMEOUT };
-};
+import { validateWaiterOptions } from "./utils";
+import { WaiterOptions, WaiterResult, waiterServiceDefaults, WaiterState } from "./waiter";
 
 const abortTimeout = async (abortSignal: AbortSignal): Promise<WaiterResult> => {
   return new Promise((resolve) => {
@@ -24,7 +19,7 @@ const abortTimeout = async (abortSignal: AbortSignal): Promise<WaiterResult> => 
  *
  * @internal
  */
-export const createWaiter = async <Client extends SmithyClient, Input>(
+export const createWaiter = async <Client, Input>(
   options: WaiterOptions<Client>,
   input: Input,
   acceptorChecks: (client: Client, input: Input) => Promise<WaiterResult>

--- a/packages/util-waiter/src/poller.ts
+++ b/packages/util-waiter/src/poller.ts
@@ -1,5 +1,5 @@
 import { sleep } from "./utils/sleep";
-import { ResolvedWaiterOptions, SmithyClient, WaiterResult, WaiterState } from "./waiter";
+import { ResolvedWaiterOptions, WaiterResult, WaiterState } from "./waiter";
 
 /**
  * Reference: https://awslabs.github.io/smithy/1.0/spec/waiters.html#waiter-retries
@@ -19,10 +19,10 @@ const randomInRange = (min: number, max: number) => min + Math.random() * (max -
  * @param input client input
  * @param stateChecker function that checks the acceptor states on each poll.
  */
-export const runPolling = async <T extends SmithyClient, S>(
-  { minDelay, maxDelay, maxWaitTime, abortController, client }: ResolvedWaiterOptions<T>,
-  input: S,
-  acceptorChecks: (client: T, input: S) => Promise<WaiterResult>
+export const runPolling = async <Client, Input>(
+  { minDelay, maxDelay, maxWaitTime, abortController, client }: ResolvedWaiterOptions<Client>,
+  input: Input,
+  acceptorChecks: (client: Client, input: Input) => Promise<WaiterResult>
 ): Promise<WaiterResult> => {
   let currentAttempt = 1;
   const waitUntil = Date.now() + maxWaitTime * 1000;

--- a/packages/util-waiter/src/utils/validate.ts
+++ b/packages/util-waiter/src/utils/validate.ts
@@ -1,10 +1,10 @@
-import { ResolvedWaiterOptions, SmithyClient } from "../waiter";
+import { ResolvedWaiterOptions } from "../waiter";
 
 /**
  * Validates that waiter options are passed correctly
  * @param options a waiter configuration object
  */
-export const validateWaiterOptions = (options: ResolvedWaiterOptions<SmithyClient>): void => {
+export const validateWaiterOptions = <Client>(options: ResolvedWaiterOptions<Client>): void => {
   if (options.maxWaitTime < 1) {
     throw `WaiterOptions.maxWaitTime must be greater than 0`;
   } else if (options.maxWaitTime <= options.minDelay) {

--- a/packages/util-waiter/src/waiter.ts
+++ b/packages/util-waiter/src/waiter.ts
@@ -1,9 +1,10 @@
-import { AbortController, Client } from "@aws-sdk/types";
+import { Client, SmithyConfiguration } from "@aws-sdk/smithy-client";
+import { AbortController, HttpHandlerOptions, MetadataBearer } from "@aws-sdk/types";
 
 /**
  * @internal
  */
-export type SmithyClient = Client<any, any, any>;
+export type SmithyClient = Client<HttpHandlerOptions, any, MetadataBearer, SmithyConfiguration<HttpHandlerOptions>>;
 export interface WaiterConfiguration<Client extends SmithyClient> {
   /**
    * Required service client

--- a/packages/util-waiter/src/waiter.ts
+++ b/packages/util-waiter/src/waiter.ts
@@ -1,11 +1,6 @@
-import { Client, SmithyConfiguration } from "@aws-sdk/smithy-client";
-import { AbortController, HttpHandlerOptions, MetadataBearer } from "@aws-sdk/types";
+import { AbortController } from "@aws-sdk/types";
 
-/**
- * @internal
- */
-export type SmithyClient = Client<HttpHandlerOptions, any, MetadataBearer, SmithyConfiguration<HttpHandlerOptions>>;
-export interface WaiterConfiguration<Client extends SmithyClient> {
+export interface WaiterConfiguration<Client> {
   /**
    * Required service client
    */
@@ -22,7 +17,7 @@ export interface WaiterConfiguration<Client extends SmithyClient> {
   abortController?: AbortController;
 }
 
-export interface WaiterOptions<Client extends SmithyClient> extends WaiterConfiguration<Client> {
+export interface WaiterOptions<Client> extends WaiterConfiguration<Client> {
   /**
    * The minimum amount of time to delay between retries in seconds. This value defaults
    * to 2 if not specified. If specified, this value MUST be greater than or equal to 1
@@ -49,7 +44,7 @@ export const waiterServiceDefaults = {
 /**
  * @private
  */
-export type ResolvedWaiterOptions<Client extends SmithyClient> = WaiterOptions<Client> &
+export type ResolvedWaiterOptions<Client> = WaiterOptions<Client> &
   Required<Pick<WaiterOptions<Client>, "minDelay" | "maxDelay">>;
 
 export enum WaiterState {

